File: mobile/ios/Sources/RoyalPMSMobile/Models/Room.swift
import Foundation

struct Room: Identifiable, Codable, Equatable {
    let id: String
    let roomNumber: String
    let floor: Int
    let category: String
    let seaView: Bool
    let status: String
    let housekeepingStatus: String
    let maintenanceNotes: String?
    let lastCleanedAt: String?
    let isVirtual: Bool
    let updatedAt: String?

    enum CodingKeys: String, CodingKey {
        case id, floor, category, status
        case roomNumber       = "room_number"
        case seaView          = "sea_view"
        case housekeepingStatus = "housekeeping_status"
        case maintenanceNotes = "maintenance_notes"
        case lastCleanedAt    = "last_cleaned_at"
        case isVirtual        = "is_virtual"
        case updatedAt        = "updated_at"
    }

    var isBlocked: Bool {
        status == "maintenance" || housekeepingStatus == "out_of_order"
    }

    var statusLabel: String {
        switch status {
        case "available":   return "Disponivel"
        case "occupied":    return "Ocupada"
        case "maintenance": return "Manutencao"
        case "reserved":    return "Reservada"
        default:            return status
        }
    }

    var housekeepingLabel: String {
        switch housekeepingStatus {
        case "clean":       return "Limpa"
        case "dirty":       return "Suja"
        case "inspected":   return "Inspecionada"
        case "out_of_order": return "Bloqueada"
        default:            return housekeepingStatus
        }
    }
}

File: mobile/android/app/src/main/java/com/royalpms/mobile/data/models/Room.kt
package com.royalpms.mobile.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Room(
    val id: String,
    @SerialName("room_number") val roomNumber: String,
    val floor: Int,
    val category: String,
    @SerialName("sea_view") val seaView: Boolean = false,
    val status: String,
    @SerialName("housekeeping_status") val housekeepingStatus: String,
    @SerialName("maintenance_notes") val maintenanceNotes: String? = null,
    @SerialName("last_cleaned_at") val lastCleanedAt: String? = null,
    @SerialName("is_virtual") val isVirtual: Boolean = false,
    @SerialName("updated_at") val updatedAt: String? = null,
)

val Room.isBlocked get() = status == "maintenance" || housekeepingStatus == "out_of_order"

val Room.statusLabel get() = when (status) {
    "available"   -> "Disponivel"
    "occupied"    -> "Ocupada"
    "maintenance" -> "Manutencao"
    "reserved"    -> "Reservada"
    else          -> status
}

val Room.housekeepingLabel get() = when (housekeepingStatus) {
    "clean"       -> "Limpa"
    "dirty"       -> "Suja"
    "inspected"   -> "Inspecionada"
    "out_of_order" -> "Bloqueada"
    else          -> housekeepingStatus
}
